### PR TITLE
Move A^+A projection out of C++

### LINKED ISSUE #44: Remove unecessary check
We probably don't need this check

https://github.com/equinor/iterative_ensemble_smoother/blob/209e93f51aa8dcf4550a71dbca0b0d62a5509706/src/iterative_ensemble_smoother/ies.cpp#L428

### PR DESCRIPTION
**Pull request by dafeda**
Resolves #44 

Removes the need to pass A to ES.

### CODE REVIEW

File: tests/test_utils.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+rng = np.random.default_rng()
+
+from iterative_ensemble_smoother.utils import _compute_AA_projection
+
+
+def test_that_svd_projection_is_same_as_pinv():
+    N = 10
+    p = 5
+    X = rng.standard_normal(size=(p, N))
+    A = X - X.mean(axis=1, keepdims=True)
```
Comment (pinkwah) [inline tests/test_utils.py:12]: Use words or comment what these symbols mean.

Comment (dafeda) [reply tests/test_utils.py:12]: Do have suggesting for good words or comments to use to describe a matrix of numbers?

File: src/iterative_ensemble_smoother/_iterative_ensemble_smoother.py [hunk 2]
```diff
@@ -109,6 +110,10 @@ def update_step(
 
         init_update(self._module_data, ensemble_mask, observation_mask)
 
+        if projection and (A.shape[0] <= A.shape[1] - 1):
```
[the remark this answers is not in the document]
Comment (dafeda) [reply src/iterative_ensemble_smoother/_iterative_ensemble_smoother.py:113]: Not sure what to write.
Run projection if you want to and if you adhere to the theory as described in the paper by Evensen?

[the remark this answers is not in the document]
Comment (dafeda) [reply src/iterative_ensemble_smoother/_iterative_ensemble_smoother.py]: My bad, it should have been `A.shape[0] < A.shape[1] - 1`

File: tests/test_utils.py (hunk 1 above)
Comment (pinkwah) [reply tests/test_utils.py:12]: Maybe at least replace `N` with `realizations` and `p` with `parameters`.

**Review by pinkwah — APPROVED** [review]

File: tests/test_utils.py (hunk 1 above)
Comment (dafeda) [reply tests/test_utils.py]: Done

**Pull request merged**